src/drivers/fluid_midikit2.cpp

```diff
@@ -0,0 +1,161 @@
+/* Haiku MidiKit2 Driver for FluidSynth
+ *
+ * Copyright (C) 2022  Cacodemon345
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA
+ */
+
+/* Parts of code taken from the CoreMIDI driver. */
+
+extern "C" {
+#include "fluid_midi.h"
+#include "fluid_mdriver.h"
+#include "fluid_settings.h"
+}
+
+#include <MidiKit.h>
+#include <midi2/MidiRoster.h>
+#include <midi2/MidiConsumer.h>
+#include <midi2/MidiProducer.h>
+#include <SupportKit.h>
+
+typedef struct
+{
+    fluid_midi_driver_t driver;
+    fluid_midi_parser_t* parser;
+
+    BMidiLocalConsumer *m_consumer;
+    int autoconn_inputs;
+} fluid_midikit2_driver_t;
+
+class fluid_midikit2_input : public BMidiLocalConsumer
+{
+    void Data(uchar *data, size_t length, bool atomic, bigtime_t time) override
+    {
+        if (atomic && callback)
+        {
+            fluid_midi_event_t *event = NULL;
+            snooze_until(time, B_SYSTEM_TIMEBASE);
+            for (int i = 0; i < length; i++)
+            {
+                event = fluid_midi_parser_parse(callback->parser, data[i]);
+            }
+            if (event)
+            {
+                callback->driver.handler(callback->driver.data, event);
+            }
+        }
+    }
+public:
+    fluid_midikit2_driver_t* callback;
+};
+
+void fluid_midikit2_driver_settings(fluid_settings_t *settings)
+{
+    fluid_settings_register_str(settings, "midi.midikit2.id", "pid", 0);
+}
+
+static void fluid_coremidi_autoconnect(fluid_midikit2_driver_t *dev)
```

Comment:
Wrong function name.

```suggestion
static void fluid_midikit2_autoconnect(fluid_midikit2_driver_t *dev)
```

Reply:
Fixed.